ibm/data_source_ibm_database.go

```diff
@@ -0,0 +1,429 @@
+package ibm
+
+import (
+	"fmt"
+	"github.com/IBM-Cloud/bluemix-go/bmxerror"
+	"github.com/IBM-Cloud/bluemix-go/models"
+
+	"github.com/IBM-Cloud/bluemix-go/api/icd/icdv4"
+	"github.com/IBM-Cloud/bluemix-go/api/resource/resourcev1/controller"
+	"github.com/IBM-Cloud/bluemix-go/api/resource/resourcev1/management"
+
+	"github.com/hashicorp/terraform/flatmap"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func dataSourceIBMDatabaseInstance() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceIBMDatabaseInstanceRead,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Description: "Resource instance name for example, my Database instance",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+
+			"resource_group_id": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The id of the resource group in which the Database instance is present",
+			},
+
+			"location": {
+				Description: "The location or the environment in which Database instance exists",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+
+			"service": {
+				Description: "The name of the Cloud Internet database service",
+				Type:        schema.TypeString,
```

Comment:
This should be optinal

Reply:
'Service' is only optional. Name is the primary key for the Resource Controller and other values can then be used to narrow it down.